src/compiler/checker.ts

```diff
@@ -17697,9 +17718,8 @@ namespace ts {
         }
 
         function isValidSpreadType(type: Type): boolean {
-            return !!(type.flags & (TypeFlags.AnyOrUnknown | TypeFlags.NonPrimitive) ||
+            return !!(type.flags & (TypeFlags.AnyOrUnknown | TypeFlags.NonPrimitive | TypeFlags.Object | TypeFlags.InstantiableNonPrimitive) ||

```

Comment:
In this `InstantiableNonPrimitive` case, shouldn't this be checking that the constraint of `type`, if it exists, `isValidSpreadType`?